hw/ip/flash_ctrl/doc/dv/index.md

```diff
@@ -87,35 +87,121 @@ All test sequences reside in `hw/ip/flash_ctrl/dv/env/seq_lib`.
 The `flash_ctrl_base_vseq` virtual sequence is extended from `cip_base_vseq` and serves as a starting point.
 All test sequences are extended from `flash_ctrl_base_vseq`.
 It provides commonly used handles, variables, functions and tasks that the test sequences can simple use / call.
-Some of the most commonly used tasks / functions are as follows:
-* task 1:
-* task 2:
+Some of the most commonly used tasks / functions are as follows: From `hw/ip/flash_ctrl/dv/env/seq/flash_ctrl_base_vseq.sv`,
+* reset_flash
+  Reset flash controller and initialize flash device using back door interface.
+* flash_ctrl_start_op
+  Start operation on the flash controller
+* flash_ctrl_write
+  Send data to `prog_fifo`. This task is used for `program operation` from controller combined with `flash_ctrl_start_op`.
+* flash_ctrl_read
+  Read data from `rd_fifo`. This task is used for `read operation` from controller combined with `flash_ctrl_start_op`.
+* wait_flash_op_done
+  Polling `op_status` until op_status.done is set.
+* do_direct_read
+  Task to read flash from the host interface. Transaction size is 4 byte per transaction.
+* flash_ctrl_intr_read
+  Task to program flash with interrupt mode.
+* flash_ctrl_intr_write
+  Task to read flash with interrupt mode.
+* send_rma_req
+  Task to initiate rma request. Once rma started, task polls `rma ack` until it completes.
 
 #### Functional coverage
 To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
 The following covergroups have been developed to prove that the test intent has been adequately met:
-* cg1:
-* cg2:
-
+`hw/ip/flash_ctrl/dv/env/flash_ctrl_env_cov.sv`
+* control_cg
+  Collects operation types, partition and cross coverage of both.
+* erase_susp_cg
+  Check if request of erase suspension occured.
+* msgfifo_level_cg
+  Covers all possible fifo status to generate interrupt for read / program.
+* rd_buff_evict_cg
+  Check sequence of operation to occur read eviction.
+* eviction_cg
+  Check whether eviction happens at all 4 caches with write / erase operation.
+  Also check each address belongs to randomly enabled scramble and ecc.
+* error_cg
+  Check errors degined in error code registers.
 ### Self-checking strategy
 #### Scoreboard
-The `flash_ctrl_scoreboard` is primarily used for end to end checking.
-It creates the following analysis ports to retrieve the data monitored by corresponding interface agents:
-* analysis port1:
-* analysis port2:
-<!-- explain inputs monitored, flow of data and outputs checked -->
+The `flash_ctrl_scoreboard` is primarily used for csr transaction integrity.
+Test bench also maintain reference momory model (associative array) per partition.
+The reference model is updated on each operation and used for expected partition values
```

Comment:
The md style recommendation is to have a single and full sentence per line: https://docs.opentitan.org/doc/sg/markdown_usage_style/#general-markdown-style.
This applies to many other sentences in this doc.

Reply:
updated